zoos.js

```diff
@@ -0,0 +1,155 @@
+const express = require('express');
+const db = require('./data/db');
+
+const server = express.Router();
+
+server.post('/', async (req, res) => {
+
+  const { name } = req.body;
+
+  if (!name) {
+
+    res.status(400).json({message: 'INVALID REQUEST: Body requires name property!'});
+    return;
+
+  }
+
+  try {
+
+    const id = await db.insert({ name }).into('zoos');
+    const newObj = await db.select().from('zoos').where('id', id[0]);
+
+    res.status(201).json(newObj[0]);
+
+  }
+
+  catch (err) {
+
+    res.status(400).json({message: 'DATABASE ERROR: Names must be unique'});
+
+  }
+
+});
+
+server.get('/', async (req, res) => {
+
+  const perPage = req.query.pageLength || 10;
+  const page = req.query.page || 1;
+
+  try {
+
+    const zoos = await db.select().from('zoos').paginate(perPage, page);
+
+    res.status(200).json(zoos);
+
+  }
+
+  catch (err) {
+
+    res.status(500).json({message: 'INTERNAL SERVER ERROR'});
+
+  }
+
+});
+
+server.get('/:id', async (req, res) => {
+
+  try {
+
+    const zoo = await db.select().from('zoos').where('id', req.params.id);
+
+    if (zoo.length) {
+
+      res.status(200).json(zoo[0]);
```

Comment:
An alternative here that might look slightly cleaner, especially with your variable name not being plural:
```javascript
const [ zoo ] = await db.select().from('zoos').where('id', req.params.id);
  if (zoo) {
    res.status(200).json(zoo);
    // ...
```
and, also, if you destructure `id`, you can use shorthand notation with an object (which is more useful when you use the variable multiple times, but it still does make that first line shorter and a bit cleaner):
```javascript
const [ zoo ] = await db.select().from('zoos').where({ id });
  if (zoo) {
    res.status(200).json(zoo);
    // ...
```